src/Infrastructure/DependencyInjection.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Application.Common.Interfaces;
+using Infrastructure.Database.Repositories;
+using Infrastructure.Database.Services;
+using Mapster;
+using Infrastructure.Clients.StockClient;
+using Infrastructure.Clients.StockClient.Configuration;
+
+namespace Infrastructure;
+
+public static class DependencyInjection
+{
+    public static IServiceCollection AddInfrastructure(this IServiceCollection services, IConfigurationSection configuration)
+    {
+        services.AddSingleton<IDateTime, DateTimeService>();
+        services.AddSingleton<IProductRepository, ProductRepository>();
+        services.AddSingleton<IStockClient>(provider 
+            => new StockClient(
```

Comment:
Placing lambda arrow after a newline causing weird indentation dilemmas for the following lines (lambda arrow and space = 3 characters, so how much should we indent the next line while keeping it divisible by 4)? After experimenting with different ideas I've found that lambda arrow on the end of previous line is the least troublesome. I don't mind to force it, but I would like an explicit decision.